rllib/execution/parallel_requests.py

```diff
@@ -143,13 +151,17 @@ def asynchronous_parallel_requests(
     for obj_ref in ready:
         remote_requests_in_flight[remote_to_actor[obj_ref]].remove(obj_ref)
 
-    # Do one ray.get().
-    results = ray.get(ready)
+    if not return_result_obj_ref_ids:
```

Comment:
nit: I think it would be cleaner to not make the type of the output depend on a flag but have distinct functions. One for getting object refs of the ready requests and this one "asynchronous_parallel_requests", which utilizes the former one.

Reply:
I'm not sure if I understand what you mean fully

Reply:
We have two return types for this function. The function really does two different things depending on a flag and I think it would be neat to have one function that returns the object refs and another one that utilizes the first one to get the ready requests themselves. Might also be a matter of taste :)

Reply:
Its pretty pythonic for a function to have multiple return times IIUC. Id prefer leaving it this way to reduce code redundancy